docs/v1/api.md

```diff
@@ -0,0 +1,333 @@
+- On `ServiceInstance` deletion, the controller must do a deprovision. The
+  controller should not delete any existing `ServiceInstanceClaim`s or
+  `ServiceInstanceBinding`s that reference the deleted `ServiceInstance`
+- On `ServiceInstanceClaim` creation:
+  - If the claim references an existing `ServiceInstance`, the controller must
+    do a bind
+  - If the claim references an existing `ServiceClass`, the controller must do
+    a provision, then bind. The controller should not create a
+    publicly-accessible `ServiceInstance`. This ensures that the
+    provisioned/bound resource will be single-tenant and dedicated
+- On `ServiceInstanceClaim` deletion:
+  - If the claim referenced an existing `ServiceInstance`, the controller must
+    do an unbind
+  - If the claim referenced an existing `ServiceClass`, the controller must do
+    an unbind, then a deprovision
+- Any successful provision operation must change the `status` field of the
+  applicable `ServiceInstanceClaim` to `provisioned`
+- Any successful bind operation must:
+  - Change the `ServiceInstanceClaim`'s `status` field to `bound`
+  - Create the appropriate resources (`Secret`s, `ConfigMap`s) to hold
+    credential and other resource-specific bind information
+  - Create a new `ServiceInstanceBinding`:
+    - With the same name as the claim’s `targetBinding` field
+    - In the same namespace as the claim itself
+    - With fields that point to all the previously created resources
+- Any successful unbind operation must:
+  - Change the claim’s `status` field to `unbound`
+  - Delete the `ServiceInstanceBinding` that was created in the bind operation
+  - Delete all resources that the aforementioned `ServiceInstanceBinding`
+    points to
+- Any failed operation must change the status field of the
+  `ServiceInstanceClaim` to `failed`, and should write the reason of the
+  failure into the `statusDescription` field
+
+The below diagram shows all of the interactions that all service catalog
+controllers are expected to have.
+
+![Complete System](./img/entire-flow.png)
+
+# Visibility
+
+The visibility of each resource and system described above is important to the
+functionality and ease-of-use in this system.
+
+The aforementioned Kubernetes resources share many of the same visibility rules
+as those in the persistent volumes and ingress systems.
+
+- `ServiceClasse`s are cluster-global (just as `StorageClass`es are)
+- `ServiceInstance`s are cluster-global (just as `PersistentVolume`s are)
+- `ServiceInstanceClaim`s are namespace-scoped (just as
+  `PersistentVolumeClaim`s are)
+- `ServiceInstanceBinding`s are namespace-scoped
+- Service catalog controllers may run on any number of namespaces, including
+  cluster-global
+
+# Security
+
+We expect that many cluster operators will need to restrict usage of many
+provisionable and bindable resources. We provide several mechanism for
+implementing these restrictions:
+
+## `ProvisioningPolicy`
```

Comment:
RBAC is described here:
http://kubernetes.io/docs/admin/authorization/
https://github.com/kubernetes/kubernetes/blob/master/pkg/apis/rbac/v1alpha1/types.go#L49

I *think* it is possible to specify who can access a specific API object (deliberately avoiding ambiguous terms like "resource") using RBAC, by specifying:

* a RoleBinding with a list of subjects who can access the object
* that refers to a Role
* that has a PolicyRule that specifies the verbs, api group, resource type, and resource name of the object

Cumbersome perhaps, since RBAC clearly wasn't designed for per-object access control, but seemingly expressible.